### ISSUE TITLE
Missing Fonts (text displayed as boxes)

### ISSUE THREAD

**Issue by cody** [fcole90/fah-snap#12]
Potential font issue on Manjaro GNOME. Font shows up as blank boxes. Interface is still functional, and folding seems to be progressing just fine based on CPU usage. Screenshot attached.
![Screenshot from 2020-10-31 17-35-46](https://user-images.githubusercontent.com/69480465/97790554-dad3dd80-1b9f-11eb-9d0f-c9ec02b72308.png)

**Comment by fcole90** [comment]
Hi, and thanks for reporting. This looks like an issue with Snap, as it happened in the past with other snaps as well. Was this your first run? Can you try to close and reopen it and see if it fixes the issue?

**Comment by cody** [comment]
No, thank *you* for working on this project! I figured it was just a Snap
issue. The issue was present on my first run, but I didn’t make the report
until probably the 4th or 5th run. The issue is persistent across
application and system restarts, as well as application reinstalls via the
graphical package manager. I’m familiar enough with the interface to still
use the software—I just can’t read any text. However, the FAH Viewer app
works perfectly, so once I start the fold via the currently unlabeled
button via the Controller app, I’m able to monitor progress with no issues
via the Viewer.

**Comment by fcole90** [comment]
Thanks for the additional details. I can try to see what's going on in the coming weeks. Unfortunately, I'm a quite busy at the moment and can take some time, but I'll take care of this

**Comment by fcole90** [comment]
Hi, I tried removing the snap from my system and installing it cleanly, to try to reproduce the issue. Unfortunately it doesn't happen for me. Can you try to do the same, so to remove it, reinstall it and then launch it from the terminal with the following command `folding-at-home-fcole90.FAHControl`?

Also, are you using any custom font from your user directory or is it installed system wide?

**Comment by nicefisher** [comment]
I can confirm this is also happening on my system as well. I may be able to help though, I installed it using  `sudo snap install folding-at-home-fcole90`

**Comment by fcole90** [comment]
Thanks for letting me know.
Are you using any custom font? 
Also, can you run it from terminal and send me the output? 😊

**Comment by nicefisher** [comment]
1: No, I just installed Manjaro and wanted to run FAH for my team, I installed other fonts after to see if it would work but to no avail.
2: After running `folding-at-home-fcole90.FAHControl` it gives me `Loading theme 'Default'
`

**Comment by fcole90** [comment]
Thanks for updating :blush: I'll try Manjaro on a VM, I wonder if there's anything specifically related to that :thinking:

**Comment by fcole90** [comment]
Which version do you have? So I'm sure to test on something as close as possible

**Comment by fcole90** [comment]
Does the following workaround fix the issue? (it should delete the font configuration of the snap)
```
rm -r ~/snap/folding-at-home-fcole90/common/.cache/fontconfig
touch ~/snap/folding-at-home-fcole90/common/.cache/fontconfig
```

It seems that's a problem on some snaps but not all, I found the workaround in this thread: https://forum.snapcraft.io/t/snapped-app-not-loading-fonts-on-fedora-and-arch/12484/87

**Comment by nicefisher** [comment]
I just installed Manjaro yesterday, so i think XCFE 20.4 or 20.2?
I ran the commands but the boxes still remain.

**Comment by fcole90** [comment]
Thanks, I'm going to investigate this deeper, I hope I can find a solution, or at least some workaround

**Comment by fcole90** [comment]
Can you additionally try this?
```
sudo rm /var/cache/fontconfig/*
rm ~/.cache/fontconfig/*
fc-cache -r
```
and then eventually reboot

It seems the issue revolves around font cache, but it's not clear how and why, and the situation changes on different snaps and systems...

**Comment by nicefisher** [comment]
I ran the commands, rebooted, no luck.

**Comment by fcole90** [comment]
Ok, I try to see what I can find :blush: Thanks for your collaboration :blush:

**Comment by nicefisher** [comment]
Cool, thanks!

**Comment by nicefisher** [comment]
I found a workaround - if you go into the Manjaro software&updates there is a AUR package for a folding at home control app, installing and running it lets me see the controller. It connects with your client that was installed so I can use it for now. Thank you though!

**Comment by fcole90** [comment]
Great to hear you could achieve that 😃

**Comment by PVasileff** [comment]
Same issue on fedora 33 + plasma (snapd-2.47.1-1.fc33.x86_64) in spotify app (when you try to change your photo on spotify)

**Comment by fcole90** [comment]
@PVasileff do you mean it happens both on FaH and on Spotify? 😊

**Comment by PVasileff** [comment]
Yes. If you installed spotify via snapd on fedora 33, after login in your spotify account - everything is ok with locales but if you go to your avatar and try to choose new image from local files - window fonts are broken:

![spotify-snapd](https://user-images.githubusercontent.com/2411033/103142311-e8be5e80-4709-11eb-8e1e-5595dd6ed295.png)

Currently issue here is for your software FaH, but spotify has same issue.. :) 
May be I need to report that and on their community.

**Comment by fcole90** [comment]
As it has become clear that the issue is not in this snap, but in the snapd daemon, please report about your experience in the snapcraft forum: https://forum.snapcraft.io/t/snapped-app-not-loading-fonts-on-fedora-and-arch/12484/90

**Comment by fcole90** [comment]
Hi, I recently updated the base snap. Can you check if you still experience such issue?
